Edit zkEVM - effective gas docs (#396)
0xPolygon/polygon-docs: docs/zkEVM/architecture/effective-gas/zkevm-egp-strat.md

    The user's signed gas price is checked against either some breakeven factor or the gas price suggested to the user.

- **Establish a criterion for when to execute transactions with user's signed gas price**
    
    Some users' signed gas prices may be sufficiently high for the user to deserve some savings. In such cases the sequencer can execute transactions with a much lower gas price.

    Hence, there's a need for a criterion that determines whether a transaction gets executed with the user's signed gas price or the effective gas price according to the RPC-estimated gas price.

- **Set a criterion for when to execute transactions with RPC-estimated EGP**
    
    The effective gas price computed with the RPC-estimated gas price could be a lot higher than the actual gas price, which is computed with the current state. In this case the user deserves further savings.

    The strategy here is to have a criterion that determines whether a transaction should get executed with the RPC-estimated effective gas price (EEGP) or the new effective gas price (NEGP), which results from the actual gas price.

- **Checking whether transactions include special opcodes**
    

>     The presence of opcodes such as the $\texttt{GASPRICE}$ and $\texttt{BALANCE}$ opcodes in transactions can cost more than expected.

Reviewer: ```suggestion
    The presence of opcodes such as $\texttt{GASPRICE}$ and $\texttt{BALANCE}$ in transactions can result in higher gas usage.
```

>     The zkEVM's approach to dealing with such transactions is to execute them with the user's signed gas price.

Reviewer: ```suggestion
    zkEVM executes such transactions with the user's signed gas price.
```

- **Enhancing prioritization of transactions**
    

>     Since transactions are sequenced according to the value they carry, with high preference given to large values, users need to provision sufficient gas price to allow for prioritization of transactions according to their needs.

Reviewer: ```suggestion
    Since transactions are sequenced in decreasing order of the specified gas price, with higher preference given to large values, users need to provision sufficient gas price to allow for prioritization of transactions according to their needs.
```
